frontend/src/scenes/dashboard/dashboardTemplates/dashboardTemplateLogic.tsx

```diff
@@ -0,0 +1,206 @@
+import { actions, afterMount, connect, kea, listeners, path } from 'kea'
+
+import type { dashboardTemplateLogicType } from './dashboardTemplateLogicType'
+import { loaders } from 'kea-loaders'
+import {
+    DashboardTemplateListing,
+    DashboardTemplateRefresh,
+    DashboardTemplateScope,
+    DashboardType,
+    FilterType,
+    Tileable,
+} from '~/types'
+import api from 'lib/api'
+import { lemonToast } from 'lib/components/lemonToast'
+import { prompt } from 'lib/logic/prompt'
+import { teamLogic } from 'scenes/teamLogic'
+import posthog from 'posthog-js'
+import { delay } from 'lib/utils'
+import { Spinner } from 'lib/components/Spinner/Spinner'
+import { Animation } from 'lib/components/Animation/Animation'
+import { AnimationType } from 'lib/animations/animations'
+import { DelayedContent } from 'lib/components/DelayedContent/DelayedContent'
+
+type TextTilePayload = {
+    type: 'TEXT'
+    body: string
+} & Tileable
+
+type InsightTilePayload = {
+    type: 'INSIGHT'
+    name: string
+    description: string
+    filters: Partial<FilterType>
+} & Tileable
+
+type TilePayload = InsightTilePayload | TextTilePayload
+
+export interface DashboardTemplateRequest {
+    dashboard_name: string
+    dashboard_description: string
+    tags: string[]
+    tiles: TilePayload[]
+}
+
+export async function pollTemplateRefreshStatus(task_id: string): Promise<void> {
```

Comment:
This feels like a lot of complexity. Imagine time goes on and we have 30 dashboard templates. I understand in that case why we need to do this in a task, but... why would we want to automatically download and install all of them? I imagined the "repository" to work similar to plugins -> you get a list of templates, and click "download" next to the templates you care about. You know, like a repository :).

And in this case, if the only thing the backend does is to refresh one JSON file (and store it in redis like we do for plugins), all this queying and fetching falls off.